### DISCUSSION TITLE
What if i want to use web3auth without "SafeEventEmitterProvider"?

### DISCUSSION THREAD

**Discussion by ankitthakur888antr** [Web3Auth/Web3Auth#1105] [Help]
If we are using Javascript for our project like in ReactJs and we want to use web3 auth then is it necessary to use typescript in that particular module? because it is using "SafeEventEmitterProvider" kind of stuff.

**Comment by Saviour1001** [comment]
It's not necessary to use TypeScript, it's just that we recommend using TypeScript so it's easier for you to build and debug. All our libraries can be operated using JS as well.
